hedera-node/hedera-evm-api/src/main/java/com/hedera/services/evm/contracts/execution/HederaEvmTxProcessor.java

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.services.evm.contracts.execution;
+
+import com.hedera.services.evm.contracts.execution.traceability.HederaEvmOperationTracer;
+import com.hedera.services.evm.store.contracts.HederaEvmMutableWorldState;
+import com.hedera.services.evm.store.contracts.HederaEvmWorldUpdater;
+import com.hedera.services.evm.store.models.HederaEvmAccount;
+import com.hederahashgraph.api.proto.java.HederaFunctionality;
+import java.time.Instant;
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.Map;
+import javax.inject.Provider;
+import org.apache.tuweni.bytes.Bytes;
+import org.hyperledger.besu.datatypes.Address;
+import org.hyperledger.besu.datatypes.Wei;
+import org.hyperledger.besu.evm.EVM;
+import org.hyperledger.besu.evm.account.EvmAccount;
+import org.hyperledger.besu.evm.frame.MessageFrame;
+import org.hyperledger.besu.evm.gascalculator.GasCalculator;
+import org.hyperledger.besu.evm.processor.AbstractMessageProcessor;
+import org.hyperledger.besu.evm.processor.ContractCreationProcessor;
+import org.hyperledger.besu.evm.processor.MessageCallProcessor;
+import org.hyperledger.besu.evm.tracing.OperationTracer;
+
+/**
+ * Abstract processor of EVM transactions that prepares the {@link EVM} and all the peripherals upon
+ * instantiation. Provides a base {@link HederaEvmTxProcessor#execute(HederaEvmAccount, Address,
+ * long, long, long, Bytes, boolean, boolean, Address)} method that handles the end-to-end execution
+ * of an EVM transaction.
+ */
+public abstract class HederaEvmTxProcessor {
+    private static final int MAX_STACK_SIZE = 1024;
+
+    protected BlockMetaSource blockMetaSource;
+    protected HederaEvmMutableWorldState worldState;
+
+    protected final GasCalculator gasCalculator;
+    // FEATURE WORK add implementation that provides logic for multiple price related methods
```

Comment:
nit: replace to do comments with tickets to capture the work

Reply:
DONE